### PR TITLE
Adding branches

### PR DESCRIPTION
**Pull request by g1st**
### Proposal
This PR includes little tweaks for general look and use. Sorry that I included them in a PR together with adding and branches functionality. I just wanted to make it look better before the meeting with the client.

### Checklist

- [x] I have made this pull request to the `master` branch
- [x] I have run `npm run lint` and there are no errors

### CODE REVIEW

File: client/src/service.js [hunk 1]
```diff
@@ -44,3 +54,13 @@ export const postJob = async (data) => {
...
 	return response.data;
 };
+
+export const postBranch = async (id, data) => {
+	const response = await post(`/branches/${id}`, data);
+	return response.data;
+};
+
+export const putBranch = async (branch_id, customer_id, data) => {
+	const response = await put(`/branches/${customer_id}/${branch_id}`, data);
```
Comment (Harriethw) [inline client/src/service.js:64]: there's some inconsistencies with our endpoints that I'd like to address next time

File: server/routes/branches.js [hunk 2]
```diff
@@ -54,9 +41,9 @@ router.get("/branches/:branch_id", async (req, res, next) => {
 	try {
 		const result = await client.query(
```
Comment (Harriethw) [inline server/routes/branches.js:42]: this isn't part of this PR, but result is assigned as a `const` twice in this function

File: server/routes/branches.js [hunk 3]
```diff
@@ -15,36 +15,23 @@ router.get("/branches", (_, res, next) => {
...
-			return res.json({ rows: result.rows });
-		} else {
-			return res.json({ rows: result.rows });
-		}
-	} catch (e) {
-		next(e);
-	} finally {
-		client.release();
-	}
+	db.query(
```
Comment (Harriethw) [inline server/routes/branches.js:21]: this is a lot cleaner, nice.

**Review by Harriethw — COMMENTED** [review]
Thanks for your hard work! What an epic PR :) Sorry I only had an hour to give it a quick look, but good progress.

File: client/src/components/CreateCustomer/Modal.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+	toggle,
+	state,
+	setState,
+	branchSaved,
+	setBranchSaved,
+}) => {
+	return (
+		<div>
+			<Modal isOpen={isOpen} toggle={toggle}>
+				<ModalHeader toggle={toggle}>Add address</ModalHeader>
```
[the remark this answers is not in the document]
Comment (g1st) [reply client/src/components/CreateCustomer/Modal.js:17]: They are needed for `reactstrap`. One of them closes modal when it's clicked anywhere outside of it (that dark zone around), the other one close it on `ESC` button press.

File: client/src/components/CreateCustomer/BranchesTable.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { Table } from "reactstrap";
+import useFetch from "../../hooks/useFetch";
+import Spinner from "../UI/Spinner";
+
+const BranchesTable = ({ state, trigger }) => {
```
[the remark this answers is not in the document]
Comment (g1st) [reply client/src/components/CreateCustomer/BranchesTable.js:7]: In the next PR we will need whole state here to prepopulate branch form for editing. But yes this one should have included only `customer_id`.

**Pull request merged**